eea/src/main/java/org/web3j/protocol/eea/crypto/PrivateTransactionDecoder.java

```diff
@@ -30,25 +35,113 @@
 
 public class PrivateTransactionDecoder {
 
-    public static RawPrivateTransaction decode(final String hexTransaction) {
+    public static LegacyPrivateTransaction decode(final String hexTransaction) {
         final byte[] transaction = Numeric.hexStringToByteArray(hexTransaction);
+        final PrivateTransactionType transactionType = getPrivateTransactionType(transaction);
+
+        if (transactionType == PrivateTransactionType.PRIVATE_1559) {
+            return decodePrivateTransaction1559(transaction);
+        }
+        return decodeLegacyPrivateTransaction(transaction);
+    }
+
+    private static PrivateTransactionType getPrivateTransactionType(final byte[] transaction) {
+        // Determine the type of the private transaction, similar to TransactionDecoder.
+        byte firstByte = transaction[0];
+        if (firstByte == PrivateTransactionType.PRIVATE_1559.getRlpType())
+            return PrivateTransactionType.PRIVATE_1559;
+        else return PrivateTransactionType.PRIVATE_LEGACY;
+    }
+
+    private static LegacyPrivateTransaction decodePrivateTransaction1559(final byte[] transaction) {
+        final byte[] encodedTx = Arrays.copyOfRange(transaction, 1, transaction.length);
+        final RlpList rlpList = RlpDecoder.decode(encodedTx);
+        final RlpList temp = (RlpList) rlpList.getValues().get(0);
+        final List<RlpType> values = temp.getValues();
+
+        System.out.println(rlpList);
```

Comment:
I assume the sysout's are temporary? You might want to mark the PR as draft?

Reply:
Fixed